src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -411,6 +415,13 @@ - (void)_handleUnsolicitedMessageFromPublisher
     os_unfair_lock_unlock(&self->_lock);
 }
 
+- (void)_handleReportBegin
+{
+    os_unfair_lock_lock(&self->_lock);
+    [self _changeState:MTRDeviceStateReachable];
```

Comment:
So... We have these various assumptions around state like "if state is reachable that means priming finished", etc.  For example, in _handleEventReport.

If we make this change, which I wholeheartedly support, we need to fix those places.

Again: can we have an actual state machine here in terms of where we are in our lifecycle, instead of inferring that from a bunch of signals like we do right now?

This part needs to get fixed for this to merge.

Separately, I wonder whether our consumers are assuming "reachable == priming done".  Home probably needs to check for that and fix things as needed if they are.  And again, exposing something more about our state here _might_ be useful.  Need to think about it.

Reply:
I am looking at changing internal logic to use a better state machine under the hood, and update the "reachability state".

At the same time I'm also trying to see if there's a smaller change we can do here - the only place where `_state` is checked other than where you pointed out in `_handleEventReport` is in where `_subscriptionAbleToReport` is used and actually makes sense. At this time I feel I could also add ReportEnd handler, and have that reconcile `_estimatedStartTime` if needed.

Reply:
I updated `_handleEventReport` logic to not depend on `_state`, and documented in comments the flow for the 4 general scenarios this logic flows through, and I can't recall why the check was there to begin with.

The other use of `_state` is actually improved by this PR (and fulfills the "TODO").